### PR TITLE
#64: Added Zero-shot model

### LINKED ISSUE #64: Support for Zero-Shot Learning models
<Feature Request>
It would be great to have ZeroShot learning models support. Given a business case, where I would like to understand a rough direction of a given dataset, this model usage at scale would be highly helpful.

### PR DESCRIPTION
**Pull request by umitbuyuksahin**
Fixes #64

### CODE REVIEW

File: tests/unit_tests/udf_wrapper_params/zero_shot/multiple_labels_single_model_single_batch.py [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+
+class MultipleLabelsSingleModelSingleBatch:
+    """
+    multiple labels, single model, single batch, batch complete
+    """
+    batch_size = 1
+    data_size = 1
+
+    input_data = [(None, "bfs_conn1", "sub_dir1", "model1",
```
Comment (tkilias) [inline tests/unit_tests/udf_wrapper_params/zero_shot/multiple_labels_single_model_single_batch.py:35]: test also multiple labels and topk>1

File: tests/unit_tests/udf_wrapper_params/zero_shot/single_bfsconn_multiple_subdir_single_model_multiple_batch.py [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    """
+    single bucketfs connection, multiple subdir, single model, multiple batch
+    """
+    batch_size = 2
+    data_size = 2
+
+    input_data = [(None, "bfs_conn1", "sub_dir1", "model1",
+                   "text1", "label1")] * data_size + \
+                 [(None, "bfs_conn1", "sub_dir2", "model2",
+                   "text2", "label2")] * data_size
```
Comment (tkilias) [inline tests/unit_tests/udf_wrapper_params/zero_shot/single_bfsconn_multiple_subdir_single_model_multiple_batch.py:38]: model names are different

File: tests/unit_tests/udf_wrapper_params/zero_shot/single_bfsconn_multiple_subdir_single_model_single_batch.py [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    """
+    single bucketfs connection, multiple subdir, single model, single batch
+    """
+    batch_size = 4
+    data_size = 2
+
+    input_data = [(None, "bfs_conn1", "sub_dir1", "model1",
+                   "text1", "label1")] * data_size + \
+                 [(None, "bfs_conn1", "sub_dir2", "model2",
+                   "text2", "label2")] * data_size
```
Comment (tkilias) [inline tests/unit_tests/udf_wrapper_params/zero_shot/single_bfsconn_multiple_subdir_single_model_single_batch.py:38]: model names are different

File: tests/unit_tests/udfs/test_zero_shot.py [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+        script_code_wrapper_function=udf_wrapper,
+        input_type="SET",
+        input_columns=[
+            Column("device_id", int, "INTEGER"),
+            Column("bucketfs_conn", str, "VARCHAR(2000000)"),
+            Column("sub_dir", str, "VARCHAR(2000000)"),
+            Column("model_name", str, "VARCHAR(2000000)"),
+            Column("text_data", str, "VARCHAR(2000000)"),
+            Column("candidate_labels", str, "VARCHAR(2000000)")
+        ],
```
Comment (tkilias) [inline tests/unit_tests/udfs/test_zero_shot.py:51]: test also with topk

**Review by tkilias — CHANGES_REQUESTED** [review]

File: exasol_transformers_extension/udfs/models/zero_shot_text_classification_udf.py [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+
+        :param model_df: Dataframe used in prediction
+        :param pred_df_list: List of predictions dataframes
+
+        :return: Prepared dataframe including input data and predictions
+        """
+        merged_df_list = []
+        for ix, pred_df in enumerate(pred_df_list):
+            merged_df = pd.merge(
+                model_df.iloc[[ix], :], pred_df, how='cross')
```
[the remark this answers is not in the document]
Comment (umitbuyuksahin) [reply exasol_transformers_extension/udfs/models/zero_shot_text_classification_udf.py:96]: This is what I wanted to do. Merging each predicted row (left_df) with its predictions(right_df).

File: tests/unit_tests/udf_wrapper_params/zero_shot/multiple_labels_single_model_single_batch.py (hunk 1 above)
Comment (umitbuyuksahin) [reply tests/unit_tests/udf_wrapper_params/zero_shot/multiple_labels_single_model_single_batch.py:35]: top_k is not a parameter for this model. This model produces predictions for each label.

File: exasol_transformers_extension/udfs/models/zero_shot_text_classification_udf.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (tkilias) [reply exasol_transformers_extension/udfs/models/zero_shot_text_classification_udf.py:96]: ah, you iterate through the rows with the for loop

File: tests/unit_tests/udf_wrapper_params/zero_shot/multiple_labels_single_model_single_batch.py (hunk 1 above)
Comment (tkilias) [reply tests/unit_tests/udf_wrapper_params/zero_shot/multiple_labels_single_model_single_batch.py:35]: ok, then I don't understand this part https://github.com/exasol/transformers-extension/blob/1306c2d5755e1a144a067568a5c9f8c61cbb6726/exasol_transformers_extension/udfs/models/zero_shot_text_classification_udf.py#L35

Comment (umitbuyuksahin) [reply tests/unit_tests/udf_wrapper_params/zero_shot/multiple_labels_single_model_single_batch.py:35]: wrong naming :+1:

Comment (tkilias) [reply tests/unit_tests/udf_wrapper_params/zero_shot/multiple_labels_single_model_single_batch.py:35]: Please, fix the naming, because it is confusing

Comment (umitbuyuksahin) [reply tests/unit_tests/udf_wrapper_params/zero_shot/multiple_labels_single_model_single_batch.py:35]: I'll do, working on size issue

**Review by tkilias — APPROVED** [review]

File: tests/fixtures/upload_language_container_fixture.py [hunk 6]
```diff
@@ -21,6 +23,10 @@ def upload_language_container(pyexasol_connection, language_container) -> str:
             container_file,
             "exasol_transformers_extension_container.tar.gz")
 
+    # remove image and build output
```
Comment (tkilias) [inline tests/fixtures/upload_language_container_fixture.py:26]:
```suggestion
    # Remove image and build output to reduce the disk usage in CI.
    # We currently, use Github Actions as the CI and its disk is limited to 14 GB.
    # TODO: This code can be removed if we moved to a CI with larger disks.
```

**Review by tkilias — CHANGES_REQUESTED** [review]

**Review by tkilias — APPROVED** [review]

**Pull request merged**